velox/exec/Window.cpp

```diff
@@ -586,7 +634,19 @@ vector_size_t Window::callApplyLoop(
           result);
       resultIndex += rowsForCurrentPartition;
       numOutputRowsLeft -= rowsForCurrentPartition;
+
+      if (currentPartition_->isPartial()) {
```

Comment:
You can move this condition into callApplyForPartitionRows itself. Then it won't repeat in this loop.

Reply:
@aditi-pandit Good suggestions. Updated.